frontend/src/pages/RunList.tsx

```diff
@@ -316,7 +316,22 @@ class RunList extends React.PureComponent<RunListProps, RunListState> {
         await this._getAndSetPipelineNames(displayRun);
 
         if (!this.props.hideExperimentColumn) {
-          await this._getAndSetExperimentNames(displayRun);
+          // Check to see if run has a provided resource reference that is an
+          // experiment that includes its name. If any condition fails, fetch
+          // the experiment manually.
+          if (displayRun.run.resource_references != null
```

Comment:
Can you extract this to a function? We should check all of the resource references, not just the one in the first index